src/boot/efi-log.c

```diff
@@ -17,6 +19,39 @@ static const uint8_t log_level_color[] = {
         [LOG_DEBUG]   = EFI_LIGHTGRAY,
 };
 
+static const char *const log_level_table[] = {
+        [LOG_EMERG]   = "emerg",
+        [LOG_ALERT]   = "alert",
+        [LOG_CRIT]    = "crit",
+        [LOG_ERR]     = "err",
+        [LOG_WARNING] = "warning",
+        [LOG_NOTICE]  = "notice",
+        [LOG_INFO]    = "info",
+        [LOG_DEBUG]   = "debug",
+};
+
+DEFINE_PRIVATE_STRING_TABLE_LOOKUP_FROM_STRING(log_level, LogLevel);
+
+int log_set_max_level(LogLevel level) {
+        assert(level >= 0 && level < _LOG_MAX);
+
+        int old = log_max_level;
+        log_max_level = level;
+        return old;
+}
+
+int log_set_max_level_from_string(const char *e) {
+        assert(e);
+        int r;
```

Comment:
Please swap the ordering and insert empty lines:
1. declare `int r;`
2. insert an empty line,
3. `assert(e)`,
4. insert an empty line again.

```c
int log_set_max_level_from_string(const char *e) {
        int r;

        assert(e);

        r = log_level_from_string(e);
        ...
```